include/boost/program_options/errors.hpp

```diff
@@ -170,7 +170,15 @@ namespace boost { namespace program_options {
         {           set_substitute("option", option_name);}
 
         std::string get_option_name() const
-        {           return get_canonical_option_name();         }
+        {           
+        	try
+        	{
+        	return get_canonical_option_name();
+        	}
+        	catch(...)
+        	{
```

Comment:
I am not quite familiar with using C++ without exceptions - what does above really accomplish? It seems to basically ignore errors, which does not seem reasonable. May I suggest that a better approach is to change get_canonical_option_prefix to use BOOST_THROW?